app/src/main/java/org/mozilla/fenix/search/SearchDialogFragment.kt

```diff
@@ -636,14 +640,16 @@ class SearchDialogFragment : AppCompatDialogFragment(), UserInteractionHandler {
                 requireContext().settings().shouldShowVoiceSearch
 
         if (isVisible) {
-            toolbarView.view.addEditActionEnd(
-                BrowserToolbar.Button(
-                    AppCompatResources.getDrawable(requireContext(), R.drawable.ic_microphone)!!,
-                    requireContext().getString(R.string.voice_search_content_description),
-                    visible = { true },
-                    listener = ::launchVoiceSearch
-                )
+            val voiceSearchAction = BrowserToolbar.Button(
+                AppCompatResources.getDrawable(requireContext(), R.drawable.ic_microphone)!!,
+                requireContext().getString(R.string.voice_search_content_description),
+                visible = { true },
+                listener = ::launchVoiceSearch
             )
+            toolbarView.view.run {
+                addEditActionEnd(IncreasedTapAreaActionDecorator(voiceSearchAction))
+                invalidateActions()
```

Comment:
@jknair0 I am wondering if `invalidateActions()` was necessary here?

Reply:
I have noticed that the bind of the actions was not getting invoked. So I had to do it manually.